log-manager/src/main/java/io/airlift/log/BufferedHandler.java

```diff
@@ -25,20 +29,33 @@
 class BufferedHandler
         extends Handler
 {
+    private static final Logger log = Logger.get(BufferedHandler.class);
     private static final int MAX_BATCH_COUNT = 1024;
     private static final byte[] POISON_MESSAGE = new byte[0];
+    private static final int MIN_WAIT_MILLIS = 100;
+    private static final int MAX_WAIT_MILLIS = 360000;
 
     private final MessageOutput messageOutput;
     private final Thread thread;
+    private Thread retryThread;
```

Comment:
Generally nobody uses raw Threads in Java anymore. Instead, you should always use an ExecutorService instead which properly decouples the concerns of task execution, from thread pool management -- and is strictly always better.